### PR TITLE
Simplify test script, and debugging

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
@bendichter A couple things in this PR.

1. Simplified main conversion script, including removing parallel run over multiple sessions (since client is more interested in other part of pipeline), making this a simple test script. Also added reliance on pathlib rather than os.
2. Debugged PR #13 (some of @alejoe91 changes broke the script).
3. Added alternative method for synchronizing with ttl pulses, reliant on [this branch/PR](https://github.com/catalystneuro/nwb-conversion-tools/pull/74). @alejoe91 let me know what you think. We may still need pickle-type interfaces for the processing stuff, but for the simple task of synchronizing the record I think it might be overkill.

@weiglszonja Note that the default nwbfile path location is now one level up.

### CODE REVIEW

File: tank_lab_to_nwb/convert_towers_task/towersnwbconverter.py [hunk 1]
```diff
@@ -16,16 +17,9 @@ class TowersNWBConverter(NWBConverter):
...
-        return self._recording_type
-
     def get_metadata(self):
-        session_path = self.data_interface_objects['VirmenData'].input_args['folder_path']
-        subject_path, session_id = os.path.split(session_path)
+        file_path = Path(self.data_interface_objects['VirmenData'].input_args['file_path'])
+        session_id = file_path.stem
 
         session_name = os.path.splitext(session_id)[0]
         date_text = [name for name in session_name.split('_') if name.isdigit()][0]
```
Comment (weiglszonja) [inline tank_lab_to_nwb/convert_towers_task/towersnwbconverter.py]: This way there is no need for `session_name`, we can get the date part directly from `session_id`
```suggestion
        date_text = [id_part for id_part in session_id.split('_') if id_part.isdigit()][0]
```

File: tank_lab_to_nwb/convert_towers_task/towersnwbconverter.py [hunk 2]
```diff
@@ -40,42 +33,13 @@ def get_metadata(self):
...
-            #         ],
-            #         'ElectricalSeries': {
-            #             'name': 'ElectricalSeries',
-            #             'description': 'raw acquisition traces'
-            #         }
-            #     }
-            # },
+            Subject=dict(),
             SpikeGLXRecording=None,
             VirmenData=dict()
```
Comment (bendichter) [inline tank_lab_to_nwb/convert_towers_task/towersnwbconverter.py]:
```suggestion
```

File: tank_lab_to_nwb/convert_towers_task/virmenbehaviordatainterface.py [hunk 3]
```diff
@@ -57,7 +57,7 @@ def get_metadata_schema(self):
 
         return metadata_schema
 
-    def convert_data(self, nwbfile: NWBFile, metadata_dict: dict, stub_test: bool = False):
+    def convert_data(self, nwbfile: NWBFile, metadata_dict: dict, stub_test: bool = False, **conversion_options):
```
Comment (bendichter) [inline tank_lab_to_nwb/convert_towers_task/virmenbehaviordatainterface.py]:
```suggestion
    def convert_data(self, nwbfile: NWBFile, metadata_dict: dict, stub_test: bool = False):
```

File: tank_lab_to_nwb/convert_towers_task/convert_towers.py [hunk 4]
```diff
@@ -1,59 +1,35 @@
...
-            #     electrode_group_metadata.update({'device_name': 'implant'})
-
-            converter.run_conversion(nwbfile_path=nwbfile_path, metadata_dict=metadata, stub_test=True)
-    else:
-        print(f"The folder ({session}) does not exist!")
+        converter = TowersNWBConverter(**input_args)
+        metadata = converter.get_metadata()
 
+        # Session specific metadata
+        metadata['NWBFile'].update(session_description="")
```
Comment (bendichter) [inline tank_lab_to_nwb/convert_towers_task/convert_towers.py]:
```suggestion
        metadata['NWBFile'].update(session_description="no description")
```

**Comment by CodyCBakerPhD** [comment]
Going to drop item (3) on the list from this per the conversation today.

**Comment by CodyCBakerPhD** [comment]
@bendichter This PR no longer has an automatic sync option since the user may have multiple methods or intentions of syncing according to the main pipeline, so we defer to that. Including the SpikeGLXRecordingExtractor in this script is really just for testing purposes, they would use an alternate method of converting it in the ipython pipeline (see nwb-conversion-tools PR's for either the pickle method or the add_extractor type function).

The rest of this PR cleans up the testing script, updates the conversion codes to nwb-conversion-tools v0.4.0, and specifies the newly added `int16` capability for writing the raw data.

File: tank_lab_to_nwb/convert_towers_task/convert_towers.py [hunk 5]
```diff
@@ -13,7 +13,8 @@
     if not nwbfile_path.is_file():
         input_args = dict(
             SpikeGLXRecording=dict(
-                file_path=spikeglx_file_path
+                file_path=spikeglx_file_path,
+                dtype="int16"
```
Comment (bendichter) [inline tank_lab_to_nwb/convert_towers_task/convert_towers.py]: I'd rather not give the user the option to change this

Comment (bendichter) [inline tank_lab_to_nwb/convert_towers_task/convert_towers.py]:
```suggestion
```

File: tank_lab_to_nwb/convert_towers_task/towersnwbconverter.py [hunk 6]
```diff
@@ -16,74 +16,38 @@ class TowersNWBConverter(NWBConverter):
...
+        vermin_file_path = Path(self.data_interface_objects['VirmenData'].input_args['file_path'])
+        session_id = vermin_file_path.stem
+        date_text = [id_part for id_part in session_id.split('_') if id_part.isdigit()][0]
         session_start = dateparse(date_text, yearfirst=True)
 
-        metadata = dict(
+        metadata = super().get_metadata()
+        metadata.update(
             NWBFile=dict(
                 identifier=session_id,
```
Comment (bendichter) [inline tank_lab_to_nwb/convert_towers_task/towersnwbconverter.py]:
```suggestion
```

**Pull request merged**